Client/Frontend/TabContentsScripts/CreditCardHelper.swift

```diff
@@ -117,9 +117,10 @@ class CreditCardHelper: TabContentScript {
 
     // MARK: Injection
 
-    func injectCardInfo(card: UnencryptedCreditCardFields,
-                        tab: Tab,
-                        completion: @escaping (Error?) -> Void) {
+    static func injectCardInfo(logger: Logger = DefaultLogger.shared,
```

Comment:
There is a few function injecting logger dependency, and logger is a variable of the class that could be injected in the init instead and avoid passing in all these functions

Reply:
Two commits added
- https://github.com/mozilla-mobile/firefox-ios/pull/14986/commits/7c42a1c48664b052885363c78d6f4900c9318f85 - Fixes issue related to logger 
- https://github.com/mozilla-mobile/firefox-ios/pull/14986/commits/3eaf810dba8f4bb8ef21947e45aab3afabd43dd2 - Fixes issue where the button to select a card was showing even when no card is present in the storage

cc @yoanarios running BR so hopefully all tests pass 🤞🏼